[newrelic-infrastructure]: Reorg values.yaml and added OpenShift parameter

bpschmitt opened the pull request:
<!--
Thank you for contributing to New Relic's Helm charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements:

* https://github.com/newrelic-experimental/helm-charts/blob/master/CONTRIBUTING.md#technical-requirements

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/newrelic-experimental/helm-charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a Github Action
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->
#### Is this a new chart

No

#### What this PR does / why we need it:

- Added OpenShift 4.x parameter which toggles /var/run/crio.sock volume instead of /var/run/docker.sock
- Also added parameter for custom CA bundle file

#### Which issue this PR fixes

N/A

#### Special notes for your reviewer:

N/A

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[mychartname]`)

paologallinaharbur commented:
There is a small issue in the values.yaml
```
charts/newrelic-infrastructure/values.yaml
  72:47     error    trailing spaces  (trailing-spaces)
```

paologallinaharbur commented:
There is still a minor syntax issue:
```
charts/newrelic-infrastructure/values.yaml
  197:31    error    no new line character at the end of file  (new-line-at-end-of-file)
```

paologallinaharbur commented on charts/newrelic-infrastructure/README.md:
```diff
@@ -44,13 +44,15 @@ This chart will deploy the New Relic Infrastructure agent as a Daemonset.
-| `discoveryCacheTTL`            | Duration since the discovered endpoints are stored in the cache until they expire. Valid time units: 'ns', 'us', 'ms', 's', 'm', 'h' | `1h` | |
+| `discoveryCacheTTL`            | Duration since the discovered endpoints are stored in the cache until they expire. Valid time units: 'ns', 'us', 'ms', 's', 'm', 'h' | `1h` |
+| `openshift` |  Enables OpenShift configuration | `false` |
+| `caBundleFile` | Path to custom CA Bundle file | |
```
caBundleFile is added only to values, but if I am not wrong it is not used in the template, is tis a typo?

paologallinaharbur commented on charts/newrelic-infrastructure/values.yaml:
```diff
@@ -68,8 +68,32 @@ serviceAccount:
+# Set openshift: true if installing in an OpenShift 4.x Cluster
+openshift: false
+
+# Openshift 4.x control plane endpoints
```
I would add some documentation explaining the different usage of this endpoints, can we configure them automatically?

bpschmitt replied:
I could add in some logic that would configure these automatically if `openshift: true`.  We'd still need to allow them to be overridden though in case non-standard endpoints are being used.  I'll rework this.

paologallinaharbur commented on charts/newrelic-infrastructure/values.yaml:
```diff
@@ -68,8 +68,32 @@ serviceAccount:
+
+# ETCD mTLS
+# https://docs.newrelic.com/docs/integrations/kubernetes-integration/installation/configure-control-plane-monitoring#mtls-how-to
+# https://docs.newrelic.com/docs/integrations/kubernetes-integration/installation/configure-control-plane-monitoring#mtls-how-to-openshift
```
same here, I would explain a bit why adding these links.

It might help to put something in the readme as well :)

bpschmitt commented:
@paologallinaharbur 

- I removed the `CABundleFile` config for now.  Will consider adding it in a future PR.
- Added some additional comments and context to the `openshift: true` parameter
- Added a comment and docs links about ETCD mTLS (the parameters were already there previously)
- Made `runAsUser` configurable.  Previously, it was statically set in the `daemonset.yaml` file.
- Updated the README with the new parameters

Let me know if there's anything else I've missed.

bpschmitt commented:
Welp, I've made a mess.  Going to close this one and start fresh instead of trying to untangle this.
